posthog/tasks/update_cache.py

```diff
@@ -51,6 +52,93 @@
 }
 
 
+def update_cached_items() -> Tuple[int, int]:
+    tasks: List[Optional[Signature]] = []
+
+    dashboard_tiles = (
+        DashboardTile.objects.filter(
+            Q(dashboard__sharingconfiguration__enabled=True)
+            | Q(dashboard__last_accessed_at__gt=timezone.now() - relativedelta(days=7))
+        )
+        .filter(
+            # no last refresh date or last refresh not in last three minutes
+            Q(last_refresh__isnull=True)
+            | Q(last_refresh__lt=timezone.now() - relativedelta(minutes=3))
+        )
+        .exclude(dashboard__deleted=True)
+        .exclude(insight__deleted=True)
+        .exclude(insight__filters={})
+        .exclude(Q(insight__refreshing=True) | Q(refreshing=True))
+        .exclude(Q(insight__refresh_attempt__gt=2) | Q(refresh_attempt__gt=2))
+        .select_related("insight", "dashboard")
+        .order_by(F("last_refresh").asc(nulls_first=True), F("insight__last_refresh").asc(nulls_first=True))
+    )
+
+    for dashboard_tile in dashboard_tiles[0:PARALLEL_INSIGHT_CACHE]:
+        tasks.append(task_for_cache_update_candidate(dashboard_tile))
+
+    shared_insights = (
+        Insight.objects.filter(sharingconfiguration__enabled=True)
+        .exclude(deleted=True)
+        .exclude(filters={})
+        .exclude(refreshing=True)
+        .exclude(refresh_attempt__gt=2)
+        .order_by(F("last_refresh").asc(nulls_first=True))
+    )
+
+    for insight in shared_insights[0:PARALLEL_INSIGHT_CACHE]:
+        tasks.append(task_for_cache_update_candidate(insight))
+
+    gauge_cache_update_candidates(dashboard_tiles, shared_insights)
+
+    tasks = list(filter(None, tasks))
+    group(tasks).apply_async()
+    return len(tasks), dashboard_tiles.count() + shared_insights.count()
+
+
+def task_for_cache_update_candidate(candidate: Union[DashboardTile, Insight]) -> Optional[Signature]:
+    candidate_tile: Optional[DashboardTile] = None if isinstance(candidate, Insight) else candidate
+    candidate_insight: Insight = candidate if isinstance(candidate, Insight) else candidate.insight
+    candidate_dashboard: Optional[Dashboard] = None if isinstance(candidate, Insight) else candidate.dashboard
+
+    try:
+        cache_key, cache_type, payload = insight_update_task_params(candidate_insight, candidate_dashboard)
+        update_filters_hash(cache_key, candidate_dashboard, candidate_insight)
```

Comment:
Changes in this file are almost all refactor and not functionality change. 

Of significance is that `update_filters_hash` is now called when processing a candidate tile or insight for cache update

So, if the filters hash is missing or out of synch with the hash of the current filters they will be updated